test/cache/json_test.rb

```diff
@@ -0,0 +1,42 @@
+require "test_helper"
+
+class Measured::Cache::JsonTest < ActiveSupport::TestCase
```

Comment:
I spent a _ton_ of time trying to write good tests for this.

I started with injecting a `StringIO` stream from test and treating it like a file. But that failed because the APIs aren't similar enough to the `File` API. I then tried both [fakefs](https://github.com/fakefs/fakefs) and [memfs](https://github.com/simonc/memfs) but both had such serious bugs they were unusable.

Ultimately I settled on putting expectations on `File` which sucks.

Reply:
> Ultimately I settled on putting expectations on File which sucks.

I'm ok with this!